internal/tls/option.go

```diff
@@ -22,48 +22,49 @@ import "crypto/tls"
 type Option func(*credentials) error
 
 var (
+	defaultConfig = &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		NextProtos: []string{
+			"http/1.1",
+			"h2",
+		},
+		CurvePreferences: []tls.CurveID{
+			tls.CurveP521,
+			tls.CurveP384,
+			tls.CurveP256,
+			tls.X25519,
+		},
+		SessionTicketsDisabled: true,
+		// PreferServerCipherSuites: true,
+		// CipherSuites: []uint16{
+		// tls.TLS_RSA_WITH_RC4_128_SHA,
+		// tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+		// tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+		// tls.TLS_RSA_WITH_AES_128_CBC_SHA256,
+		// tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+		// tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+		// tls.TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+		// tls.TLS_ECDHE_RSA_WITH_RC4_128_SHA,
+		// tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+		// tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+		// tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+		// tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,
+		// tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+		// tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, // Maybe this is work on TLS 1.2
+		// tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA, // TLS1.3 Feature
+		// tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA, // TLS1.3 Feature
+		// tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305, // Go 1.8 only
+		// tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305, // Go 1.8 only
+		// },
+		ClientAuth: tls.NoClientCert,
+	}
 	defaultOpts = []Option{
-		WithTLSConfig(&tls.Config{
```

Comment:
inlineing is better

Reply:
I use it in the test case

Reply:
I know, but you shouldn't use field variable for FOP testing.
you can overwrite configuration by FOP in testing